src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtTypesAndCodesServiceTest.java

```diff
@@ -0,0 +1,180 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.dts.fact.config.security.RolesProvider;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtDxCode;
+import uk.gov.hmcts.dts.fact.entity.CourtType;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.CourtTypesAndCodes;
+import uk.gov.hmcts.dts.fact.model.admin.DxCode;
+import uk.gov.hmcts.dts.fact.repositories.CourtDxCodesRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtTypeRepository;
+import uk.gov.hmcts.dts.fact.util.MapCourtCode;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@ContextConfiguration(classes = AdminCourtTypesAndCodesService.class)
+public class AdminCourtTypesAndCodesServiceTest {
+
+    private static final int COURT_TYPE_COUNT = 3;
+    private static final List<CourtType> COURT_TYPES = new ArrayList<>();
+    private static final String COURT_SLUG = "some slug";
+    private static final String NOT_FOUND = "Not found: ";
+    private static final String GBS_CODE = "123";
+
+    private static final List<CourtType> EXPECTED_COURT_TYPES_ENTITY = Arrays.asList(
+        new CourtType(1,"test1"),
+        new CourtType(2,"test2"),
+        new CourtType(3,"test3")
+    );
+
+    private static final List<uk.gov.hmcts.dts.fact.model.admin.CourtType> EXPECTED_COURT_TYPES = Arrays.asList(
+        new uk.gov.hmcts.dts.fact.model.admin.CourtType(1,"test1",null),
+        new uk.gov.hmcts.dts.fact.model.admin.CourtType(2,"test2",null),
+        new uk.gov.hmcts.dts.fact.model.admin.CourtType(3, "test3",null)
+    );
+
+
+
+    private static final List<DxCode> EXPECTED_COURT_DX_CODES = Arrays.asList(
+        new DxCode("Code 1",null,null),
```

Comment:
Is there ever going to be a case where the second and third parameter will not be null, if so, should we test that case too?

Reply:
updated the tests